app/src/main/java/com/cornellappdev/uplift/ui/components/goalsetting/ScrollWheel.kt

```diff
@@ -0,0 +1,242 @@
+import androidx.compose.foundation.layout.width
+import androidx.compose.foundation.lazy.LazyColumn
+import androidx.compose.foundation.lazy.rememberLazyListState
+import androidx.compose.foundation.shape.RoundedCornerShape
+import androidx.compose.material3.Text
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.LaunchedEffect
+import androidx.compose.runtime.getValue
+import androidx.compose.runtime.mutableIntStateOf
+import androidx.compose.runtime.mutableStateOf
+import androidx.compose.runtime.remember
+import androidx.compose.runtime.rememberCoroutineScope
+import androidx.compose.runtime.setValue
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.draw.clip
+import androidx.compose.ui.graphics.Color
+import androidx.compose.ui.layout.onGloballyPositioned
+import androidx.compose.ui.layout.positionInParent
+import androidx.compose.ui.platform.LocalDensity
+import androidx.compose.ui.text.TextStyle
+import androidx.compose.ui.text.font.FontWeight
+import androidx.compose.ui.tooling.preview.Preview
+import androidx.compose.ui.unit.Dp
+import androidx.compose.ui.unit.dp
+import androidx.compose.ui.unit.sp
+import com.cornellappdev.uplift.util.GRAY01
+import com.cornellappdev.uplift.util.GRAY08
+import com.cornellappdev.uplift.util.montserratFamily
+import kotlinx.coroutines.launch
+import kotlin.math.abs
+
+/**
+ * @param width: the width of the scroll wheel
+ * @param itemHeight: the height of each item in the scroll wheel
+ * @param numberOfDisplayedItems: the number of items displayed at once
+ * @param items: the list of items to display
+ * @param initialItem: the item to select initially
+ * @param itemScaleFact: the scale factor for the selected item
+ * @param textStyle: the text style for the items
+ * @param textColor: the color of the unselected items
+ * @param selectedTextColor: the color of the selected item
+ * @param onItemSelected: the callback for when an item is selected
+ * @return ScrollWheel composable that allows the user to select an item from a list of items
+ */
+@Composable
+fun <T> ScrollWheel(
+    width: Dp,
+    itemHeight: Dp,
+    numberOfDisplayedItems: Int = 3,
+    items: List<T>,
+    initialItem: T,
+    itemScaleFact: Float = 1.5f,
+    textStyle: TextStyle,
+    textColor: Color,
+    selectedTextColor: Color,
+    onItemSelected: (index: Int, item: T) -> Unit = { _, _ -> },
+) {
+    val itemHalfHeight = LocalDensity.current.run { itemHeight.toPx() / 2f }
+    val scrollState = rememberLazyListState(0)
+    var lastSelectedIndex by remember {
+        mutableIntStateOf(0)
+    }
+    var itemsState by remember {
+        mutableStateOf(items)
+    }
+    LaunchedEffect(items) {
+        var targetIndex = items.indexOf(initialItem) - 1
+        targetIndex += ((Int.MAX_VALUE / 2) / items.size) * items.size
+        itemsState = items
+        lastSelectedIndex = targetIndex
+        scrollState.scrollToItem(targetIndex)
+    }
```

Comment:
Wait this component is actually insane. Super clever implementation, working with MAX_VALUE items and then scrolling to the middle so theyd never reach the end reasonably. Super super clever, I was honestly wondering how this component would end up working myself.

I'm honestly most surprised it's running fine, even with technically `MAX_VALUE` items! I guess it's optimized to only render a handful at a time. So impressive though, really great workaround.

Reply:
Yeah, I really struggled with this component, and GPT was not cooking, so I found some custom implementation from stack overflow and repurposed it to fit our design: https://stackoverflow.com/questions/69734451/is-there-a-way-to-create-scroll-wheel-in-jetpack-compose. Definitely would not have thought about that on my own lol